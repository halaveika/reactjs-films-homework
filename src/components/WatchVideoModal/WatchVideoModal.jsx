import React, { Component } from 'react';
import { Modal } from 'antd';
import PropTypes from 'prop-types';
import HttpService from '../../modules/api/httpService';
import ModalContent from '../ModalContent';
import './WatchVideoModal.scss';

export default class WatchVideoModal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isModalVisible: false,
      video: '',
    };
    this.videoRef = React.createRef();
  }

  handleCancel() {
    this.videoRef.current.src = `${this.state.video}`;
    this.setState({ ...this.state, isModalVisible: false });
  }

  async showModal() {
    const { id } = this.props;
    const video = await HttpService.movieVideoRequest(id);
    this.setState({ ...this.state, isModalVisible: true, video });
  }

  render() {
    const { isModalVisible } = this.state;
    return (
      <>
        { React.cloneElement(this.props.children, { onClick: this.showModal.bind(this) }) }
        <Modal
          visible={isModalVisible}
          cancelText="cancel"
          okButtonProps={{ style: { display: 'none' } }}
          onCancel={this.handleCancel.bind(this)}
          closable={false}
        >
          {' '}
          <ModalContent videoRef={this.videoRef} video={this.state.video} />
        </Modal>
      </>
    );
  }
}

WatchVideoModal.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.number,
};
