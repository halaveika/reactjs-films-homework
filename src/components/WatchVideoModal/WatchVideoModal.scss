.videoIframe {
  width: 82vw;
  height: 85vh;
}
