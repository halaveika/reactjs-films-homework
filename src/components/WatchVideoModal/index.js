import WatchVideoModal from './WatchVideoModal';

export default WatchVideoModal;
