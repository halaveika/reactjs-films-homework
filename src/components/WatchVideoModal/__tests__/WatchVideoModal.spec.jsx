import React from 'react';
import { create, act } from 'react-test-renderer';
import { Card, Button } from 'antd';
import WatchVideoModal from '../WatchVideoModal';

const WatchVideoModalProps = {
  id: '2412412',
  video: 'youtube-link',
  handleVideo: jest.fn(),
};

afterEach(() => {
  jest.clearAllMocks();
});

const toggleActive = jest.fn();

describe('test WatchVideoModal component', () => {
  it('should render WatchVideoModal component', () => {
    const component = create(
      <WatchVideoModal
        video={WatchVideoModalProps.video}
        handleVideo={WatchVideoModalProps.handleVideo}
        id={WatchVideoModalProps.id}
        children={<Button onClick={toggleActive} />}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('handleVisibleChange should change state', () => {
  //   const mockCallback = jest.spyOn(NavBar.prototype, 'setState').mockImplementation();

  //   const { root } = create(
  //     <NavBar
  //       overview={mockData.overview}
  //     />,
  //   );
  //   root.instance.handleVisibleChange(true);
  //   expect(mockCallback.mock.calls.length).toBe(1);
  //   expect(mockCallback.mock.calls[0][0]).toStrictEqual({visible:true});
  // });

  // it('handleVisibleChange should handle button click', () => {
  //   const mockCallback = jest.spyOn(NavBar.prototype, 'handleVisibleChange').mockImplementation();

  //   const { root } = create(
  //     <NavBar
  //       overview={mockData.overview}
  //     />,
  //   );
  //   expect(root.instance.state.visible).toBe(false);
  //   act(() => { root.findAllByType('button')[1].props.onClick(); });
  //   expect(mockCallback.mock.calls.length).toBe(1);
  //   expect(mockCallback.mock.calls[0][0]).toBe(true);
  // });
});
