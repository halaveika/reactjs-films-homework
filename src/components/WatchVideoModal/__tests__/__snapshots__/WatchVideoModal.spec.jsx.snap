// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test WatchVideoModal component click should change state 1`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  type="button"
/>
`;

exports[`test WatchVideoModal component handleCancel should change state 1`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  type="button"
/>
`;
